packages/vite/src/node/plugins/css.ts

```diff
@@ -778,6 +779,36 @@ async function compileCSS(
   }
 }
 
+export function createHTMLCSSCompiler(config: ResolvedConfig): {
```

Comment:
Looks like the bulk of the code is referenced from [here](https://github.com/vitejs/vite/blob/f1ef7533b6174df0d1bb1f754fbc5829a095851e/packages/vite/src/node/plugins/css.ts#L142-L191), I think we can try re-using the function to remove the duplication.

Reply:
may I create `cssPlugin` instance and use transform method to transform style code ?

Reply:
I'm not sure if that's desirable, but maybe separating `createHTMLCSSCompiler` would be more understandable for now. But to be fair, vite-plugin-svelte also [does what you mentioned](https://github.com/sveltejs/vite-plugin-svelte/blob/d65f17f09cc5370a2051af14f6ff324356ddf1d1/packages/vite-plugin-svelte/src/utils/preprocess.ts#L41-L49), and it feels a bit sketchy 😅

Reply:
hhhh, I think so. if call `compileCSS` in the html and we need define more method.

Reply:
find is better than create instance.

Reply:
I actually recommended `createHTMLCSSCompiler`, but it's fine anyways 😆 I think there's a better way to tackle this in my other review comment.